File: answer/greed/gym_clothes.py
def solution(n, lost, reserve):
    student = [0] + [1] * n + [0]
    for i in lost:
        student[i] -= 1
    for i in reserve:
        student[i] += 1
    i = 1
    while i <= n:
        if (student[i] == 0):
            if (student[i - 1] == 2):
                student[i - 1] -= 1
                student[i] += 1
            elif (student[i + 1] == 2):
                student[i + 1] -= 1
                student[i] += 1
        i += 1
    for i in student:
        if i == 0:
            n -= 1
    n += 2
    return n

File: answer/greed/gym_clothes.cpp
/*
 * I suppose vectors are 'arranged' that's why I spend a lot of time to solve it.
 */

#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(int n, vector<int> lost, vector<int> reserve) {
	int answer = 0;
	int i = 0;
	int j = 0;
	sort(lost.begin(), lost.end());
	sort(reserve.begin(), reserve.end());
	while (i < lost.size() && j < reserve.size())
	{
		if (lost[i] == reserve[j])
		{
			lost.erase(lost.begin() + i);
			reserve.erase(reserve.begin() + j);
		}
		else if (lost[i] > reserve[j])
			j++;
		else
			i++;
	}
	i = 0;
	j = 0;
	while (i < lost.size() && j < reserve.size())
	{
		if (lost[i] - 1 > reserve[j])
			j++;
		else if (lost[i] - 1 == reserve[j] || \
                 lost[i] + 1 == reserve[j])
		{
			lost.erase(lost.begin() + i);
			reserve.erase(reserve.begin() + j);
		}
		else
			i++;
	}
	answer = n - int(lost.size());
	return answer;
}

/* main example
int main()
{
	vector<int> lost;
	lost.push_back(1);
	lost.push_back(3);
	lost.push_back(5);
	vector<int> reserve;
	reserve.push_back(2);
	reserve.push_back(3);
	reserve.push_back(4);
	return(printf("%d\n", solution(5, lost, reserve)));
}*/
